ClinGen/clincoded
# Connects to #672. VCI Basic Information Tab UI.

This PR consists of changes pertinent to Issue #672.

**Technical notes:**
1. Added a JSON mapping file (e.g. SO_term.json) consisting of **SO term** and **SO id** pairs. The list of mapping pairs are taken from http://uswest.ensembl.org/info/genome/variation/predicted_data.html#consequence_type_table
2. Created a new "mapping" sub-directory in the anticipation of more mapping files being required, such as NM-to-NC and NM-to-EN.

**Known issues:**
1. The sortable "All Transcripts" table columns have not yet been implemented (as requested in https://github.com/ClinGen/clincoded/issues/672#issuecomment-223381387).
2. The RefSeq transcripts have not been merged into the "All Transcripts" table with the Ensembl transcripts (as requested in https://github.com/ClinGen/clincoded/issues/672#issuecomment-223381387). This is because the logical mapping of NM-to-NC has not yet been figured out.
3. The NC value in the "Protein Change" column in the "Primary Transcript" table is temporarily using the 1st index in the NC array. This also is because the logical mapping of NM-to-NC has not yet been figured out.

**Steps to test this PR:**
1. Login and go to http://localhost:6543/variant-central/?variant=35eedfdc-2685-11e5-b6a0-60f81dc5b05a
2. In the "Basic Information" tab, expect to see data being displayed in both tables. Also clicking on various LinkOut links should result in new windows with the expected destinations.
